### PR TITLE
Added functionality to convert data to CSV/TSV and JSON and vice versa.

### PR DESCRIPTION
**Pull request by john-thuo1**
### Contributor Checklist
- [✔️] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch.
### Description
This pull request adds functionality to convert data into CSV, TSV, and JSON formats. The changes include:
- Extension of the `convert_to_json` and `convert_to_csv_or_tsv` methods in `convert.py`.
- Updates to the CLI in `main.py` to support new conversion commands.
#### Example Commands:
1. **From JSON to CSV**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./fli/French/translations.json --output-type csv --output-dir ./converted/
2. **From CSV to JSON**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./converted/French/translations.csv --output-type json --output-dir ./converted/
3. **From JSON to TSV**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./fli/French/translations.json --output-type tsv --output-dir ./converted/ 
4. **From TSV to JSON**:
   ```bash
   scribe-data convert --lan French --data-type translations --input-file ./converted/French/translations.tsv --output-type json --output-dir ./converted/ 

I tested the changes by running the CLI commands and verifying that the output matched the expected formats. I want to get feedback on the approaches before implementing the code tests and the SQLite conversion support

### Related issue

- Decouple convert from get in Scribe-Data CLI #213

### CODE REVIEW

File: src/scribe_data/cli/main.py [hunk 1]
```diff
@@ -179,22 +183,56 @@ def main() -> None:
...
         type=str,
         choices=["json", "csv", "tsv", "sqlite"],
+        required=True,
         help="The output file type.",
     )
+    convert_parser.add_argument(
+        "-od",
+        "--output-dir",
+        type=Path,
+        required=True,
```
Comment (andrewtavis) [inline src/scribe_data/cli/main.py:221]: There's no need for this to be required. Ideally the default directories would be used if this isn't included :)

**Comment by andrewtavis** [comment]
@mhmohona, if you have time, it'd be great if you could take a first look at the changes to `convert.py` :) Let me know if I should check it from the start!

**Comment by andrewtavis** [comment]
@john-thuo1, is there a way that we can get the commit history here cleaned up a bit? It's hard for us to review this as I can't tell what the changes are that are in here and already on main. What might make sense would be to make a new branch from the current version of main and add your changes to that? Then open a new PR?

**Comment by john-thuo1** [comment]
> @john-thuo1, is there a way that we can get the commit history here cleaned up a bit? It's hard for us to review this as I can't tell what the changes are that are in here and already on main. What might make sense would be to make a new branch from the current version of main and add your changes to that? Then open a new PR?

Alright. Apologies for that. Still getting the hang of cherry-picking.

**Comment by andrewtavis** [comment]
Hey it's hard stuff, @john-thuo1! Thanks for opening #338 :)

**Pull request closed without merging**